source/server/admin/admin.cc

```diff
@@ -450,7 +450,12 @@ Http::Code AdminImpl::handlerClusters(absl::string_view url,
 
 void AdminImpl::addAllConfigToDump(envoy::admin::v3::ConfigDump& dump,
                                    const absl::optional<std::string>& mask) const {
-  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+  Envoy::Server::ConfigTracker::CbsMap callbacks_map = std::move(config_tracker_.getCallbacksMap());
+  if (!server_.clusterManager().clusters().empty()) {
+    callbacks_map.emplace("endpoint", [this] { return dumpEndpointConfigs(); });
```

Comment:
since dumpEndpontConfigs is not fully completed, should we defer this and changes in addResourceToDump to  the next PR? Otherwise, anyone doing config dump will get imp-complete EDS. @htuch WDYT?

Reply:
Yeah, that's fair enough. Or you could use a runtime feature flag to control this behavior. https://github.com/envoyproxy/envoy/blob/master/CONTRIBUTING.md#runtime-guarding. Might be overkill.